tests/test_batch_reconstruction.py

```diff
@@ -96,6 +97,70 @@ def handle_async_exception(loop, ctx):
         for elem in r:
             assert type(elem) is GFElement
 
+
+@mark.asyncio
+async def test_fft():
+    n = 4
+    p = 97
+    t = 1
+    fp = GF.get(p)
+
+    # loop = asyncio.get_event_loop()
+    # loop.set_exception_handler(handle_async_exception)
+
+    # Test with simple case: n = 4, t =1
+    # After AVSS, poly1 = x + 2, poly2 = 3x + 4, secret1 = 2, secret2 = 4
+    # Hard code the shared secret value as input into batch_reconstruction function\
+    point = EvalPoint(fp, n, use_fft=True)
+    omega = point.omega
+    shared_secrets = [(omega ** 0 + 2, 3 * omega ** 0 + 4),
+                      (omega ** 1 + 2, 3 * omega ** 1 + 4),
+                      (omega ** 2 + 2, 3 * omega ** 2 + 4),
+                      (omega ** 3 + 2, 3 * omega ** 3 + 4)]
+    shared_secrets = [(a.value, b.value) for (a, b) in shared_secrets]
+    # shared_secrets = [()]
+    # Test 1: Correct decoding with all four points
+    sends, recvs = simple_router(n)
+    towait = []
+    for i in range(n):
+        ss = tuple(map(fp, shared_secrets[i]))
+        towait.append(batch_reconstruct(ss, p, t, n, i,
+                                        sends[i], recvs[i], True, use_fft=True))
+    results = await asyncio.gather(*towait)
+    for r in results:
+        assert r == [2, 4]
+
+    # Test 2: Correct decoding with up to 1 error
```

Comment:
Do you think it will be better to separate these to multiple tests if they are separate? It shouldn't happen that the first test fails and then the rest of the test isn't executed (though Python might actually execute it)

Reply:
I think so too. That makes it easier to figure out what test is actually being run. But it would be better to set up common test cases inside by subclassing pytest.TestCase, since most of these tests have common setup code.

Reply:
For now, I just looked at how the other batch reconstruction tests were being done and used the same format.

Reply:
Yea the existing code is far from perfect. Most of it was written when we were in a time crunch and it has been reviewed properly or written properly.

Reply:
Should I refactor the tests?

Reply:
Yea, it would be better to refactor them.